src/main/java/uk/gov/hmcts/reform/blobrouter/reconciliation/service/SummaryReportService.java

```diff
@@ -62,6 +62,14 @@ public void process(LocalDate date) {
 
         EnvelopeSupplierStatement envelopeSupplierStatement = optSupplierStatement.get();
 
+        var existingReports = reconciliationReportRepository
+            .findByStatementId(envelopeSupplierStatement.id);
+
+        if (existingReports.size() == TargetStorageAccount.values().length) {
```

Comment:
it would safer to also check the values or `account` are as expected (no duplicates) and log an error if not.